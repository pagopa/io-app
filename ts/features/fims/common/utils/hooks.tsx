import * as React from "react";
import { ServiceId } from "../../../../../definitions/backend/ServiceId";
import { useIODispatch, useIOSelector } from "../../../../store/hooks";
import { serviceByIdPotSelector } from "../../../services/details/store/reducers";
import { loadServiceDetail } from "../../../services/details/store/actions/details";
import { isStrictNone } from "../../../../utils/pot";

export const useAutoFetchingServiceByIdPot = (serviceId: ServiceId) => {
  const dispatch = useIODispatch();
  const serviceData = useIOSelector(state =>
    serviceByIdPotSelector(state, serviceId)
  );
  const shouldFetchServiceData = isStrictNone(serviceData);

  React.useEffect(() => {
    if (shouldFetchServiceData) {
      dispatch(loadServiceDetail.request(serviceId));
    }
  }, [dispatch, serviceId, shouldFetchServiceData]);

  return { serviceData };
};
