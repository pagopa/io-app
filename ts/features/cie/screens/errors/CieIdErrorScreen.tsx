import React from "react";
import { OperationResultScreenContent } from "../../../../components/screens/OperationResultScreenContent";
import useNavigateToLoginMethod from "../../../../hooks/useNavigateToLoginMethod";
import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import I18n from "../../../../i18n";
import { TranslationKeys } from "../../../../../locales/locales";
import { useAvoidHardwareBackButton } from "../../../../utils/useAvoidHardwareBackButton";

const CIE_PIN_DESC: TranslationKeys =
  "authentication.cie_id.error_screen.cie_pin_supported.description";
const CIE_PIN_ACTION_LABEL: TranslationKeys =
  "authentication.cie_id.error_screen.cie_pin_supported.primary_action_label";
const SPID_DESC: TranslationKeys =
  "authentication.cie_id.error_screen.cie_pin_not_supported.description";
const SPID_ACTION_LABEL: TranslationKeys =
  "authentication.cie_id.error_screen.cie_pin_not_supported.primary_action_label";

const CieIdErrorScreen = () => {
  const { navigateToIdpSelection, navigateToCiePinInsertion, isCieSupported } =
    useNavigateToLoginMethod();
  const { popToTop } = useIONavigation();

  useAvoidHardwareBackButton();

  const subtitle = I18n.t(isCieSupported ? CIE_PIN_DESC : SPID_DESC);
  const primaryActionLabel = I18n.t(
    isCieSupported ? CIE_PIN_ACTION_LABEL : SPID_ACTION_LABEL
  );
  const secondaryActionLabel = I18n.t(
    "authentication.cie_id.error_screen.secondary_action_label"
  );

  return (
    <OperationResultScreenContent
      title={I18n.t("authentication.cie_id.error_screen.title")}
      subtitle={subtitle}
      pictogram="help"
      action={{
        testID: "cie-id-error-primary-action",
        label: primaryActionLabel,
        accessibilityLabel: primaryActionLabel,
        onPress: isCieSupported
          ? navigateToCiePinInsertion
          : navigateToIdpSelection
      }}
      secondaryAction={{
        testID: "cie-id-error-secondary-action",
        label: secondaryActionLabel,
        accessibilityLabel: secondaryActionLabel,
        onPress: popToTop
      }}
    />
  );
};

export default CieIdErrorScreen;
