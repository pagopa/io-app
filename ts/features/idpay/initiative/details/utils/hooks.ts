import * as React from "react";
import * as pot from "@pagopa/ts-commons/lib/pot";
import { useIODispatch, useIOSelector } from "../../../../../store/hooks";
import {
  idpayPaginatedTimelineSelector,
  idpayTimelineCurrentPageSelector,
  idpayTimelineIsLastPageSelector,
  idpayTimelineLastUpdateSelector,
  idpayTimelineSelector
} from "../store";
import { idpayTimelinePageGet } from "../store/actions";

export const useInitiativeTimelineFetcher = (
  initiativeId: string,
  pageSize: number,
  onError: () => void
) => {
  const dispatch = useIODispatch();

  const paginatedTimelinePot = useIOSelector(idpayPaginatedTimelineSelector);
  const isLastPage = useIOSelector(idpayTimelineIsLastPageSelector);
  const currentPage = useIOSelector(idpayTimelineCurrentPageSelector);
  const lastUpdate = useIOSelector(idpayTimelineLastUpdateSelector);

  const timeline = useIOSelector(idpayTimelineSelector);

  const isLoading = pot.isLoading(paginatedTimelinePot);
  const isError = pot.isError(paginatedTimelinePot);

  const [isRefreshing, setIsRefreshing] = React.useState(false);

  React.useEffect(() => {
    if (currentPage >= 0 && isRefreshing) {
      setIsRefreshing(false);
    }
  }, [currentPage, isRefreshing]);

  const refresh = () => {
    if (!isRefreshing) {
      setIsRefreshing(true);
      fetchPage(0);
    }
  };

  const fetchPage = (page: number) =>
    dispatch(idpayTimelinePageGet.request({ initiativeId, page, pageSize }));

  const fetchNextPage = () => {
    if (isLastPage || isLoading) {
      return;
    }
    if (isError) {
      onError();
    } else {
      fetchPage(currentPage + 1);
    }
  };

  const retryFetchLastPage = () => {
    fetchPage(currentPage);
  };

  return {
    isLoading,
    isRefreshing,
    timeline,
    fetchPage,
    refresh,
    fetchNextPage,
    retryFetchLastPage,
    currentPage,
    lastUpdate
  } as const;
};
