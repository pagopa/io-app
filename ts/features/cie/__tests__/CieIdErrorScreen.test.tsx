import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import CieIdErrorScreen from "../screens/errors/CieIdErrorScreen";
import * as useNavigateToLoginMethod from "../../../hooks/useNavigateToLoginMethod";

const mockNavigateToCiePinInsertion = jest.fn();
const mockNavigateToIdpSelection = jest.fn();
const mockNavigateToCieIdScreen = jest.fn();
const mockPopToTop = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      popToTop: mockPopToTop
    })
  };
});

describe("CieIdErrorScreen where device supports NFC", () => {
  afterEach(jest.clearAllMocks);
  beforeEach(() => {
    jest.spyOn(useNavigateToLoginMethod, "default").mockImplementation(() => ({
      navigateToCiePinInsertion: mockNavigateToCiePinInsertion,
      navigateToIdpSelection: mockNavigateToIdpSelection,
      navigateToCieIdLoginScreen: mockNavigateToCieIdScreen,
      isCieSupported: true
    }));
  });

  it("Should be defined", testIsDefined);
  it("Should match the snapshot", testMatchSnapshot);
  it("Should navigate to CIE + PIN screen", () => {
    const { getByTestId } = render(<CieIdErrorScreen />);
    const primaryAction = getByTestId("cie-id-error-primary-action");

    fireEvent.press(primaryAction);

    expect(mockNavigateToCiePinInsertion).toHaveBeenCalled();
    expect(mockNavigateToIdpSelection).not.toHaveBeenCalled();
    expect(mockNavigateToCieIdScreen).not.toHaveBeenCalled();
    expect(mockPopToTop).not.toHaveBeenCalled();
  });
  it("Should properly call pop-to-top", testPopToTop);
});
describe("CieIdErrorScreen where device doesn't support NFC", () => {
  afterEach(jest.clearAllMocks);
  beforeEach(() => {
    jest.spyOn(useNavigateToLoginMethod, "default").mockImplementation(() => ({
      navigateToCiePinInsertion: mockNavigateToCiePinInsertion,
      navigateToIdpSelection: mockNavigateToIdpSelection,
      navigateToCieIdLoginScreen: mockNavigateToCieIdScreen,
      isCieSupported: false
    }));
  });

  it("Should be defined", testIsDefined);
  it("Should match the snapshot", testMatchSnapshot);
  it("Should navigate to Idp Selection screen", () => {
    const { getByTestId } = render(<CieIdErrorScreen />);
    const primaryAction = getByTestId("cie-id-error-primary-action");

    fireEvent.press(primaryAction);

    expect(mockNavigateToIdpSelection).toHaveBeenCalled();
    expect(mockNavigateToCiePinInsertion).not.toHaveBeenCalled();
    expect(mockNavigateToCieIdScreen).not.toHaveBeenCalled();
    expect(mockPopToTop).not.toHaveBeenCalled();
  });
  it("Should properly call pop-to-top", testPopToTop);
});

function testIsDefined() {
  const component = render(<CieIdErrorScreen />);

  expect(component).toBeDefined();
}

function testMatchSnapshot() {
  const component = render(<CieIdErrorScreen />);

  expect(component).toMatchSnapshot();
}

function testPopToTop() {
  const { getByTestId } = render(<CieIdErrorScreen />);
  const primaryAction = getByTestId("cie-id-error-secondary-action");

  fireEvent.press(primaryAction);

  expect(mockNavigateToCiePinInsertion).not.toHaveBeenCalled();
  expect(mockNavigateToIdpSelection).not.toHaveBeenCalled();
  expect(mockNavigateToCieIdScreen).not.toHaveBeenCalled();
  expect(mockPopToTop).toHaveBeenCalled();
}
