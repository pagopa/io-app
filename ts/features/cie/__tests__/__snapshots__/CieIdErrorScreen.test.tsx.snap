// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CieIdErrorScreen where device doesn't support NFC Should match the snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flexGrow": 1,
      "marginHorizontal": 24,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
      ]
    }
    handlerTag={6}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      [
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={120}
          bbWidth={120}
          color="#00C5CA"
          focusable={false}
          height={120}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 120,
                "width": 120,
              },
            ]
          }
          tintColor="#00C5CA"
          vbHeight={240}
          vbWidth={240}
          width={120}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="M220.46 181.2c-.11-.13-10.98-13.21-23.59-26.68-20.84-22.26-30.19-28.43-34.36-29.69-5.3-1.59-9.25-.71-11.14 2.51-1.96 3.35-1.13 8.07.92 10.06.66.63.8 1.63.35 2.42-.45.79-1.38 1.18-2.26.94-.18-.05-18.54-4.97-29.01-5.63-1.05-.07-2.06.29-2.85.99-.81.72-1.28 1.72-1.32 2.81-.09 2.07 1.39 3.83 3.43 4.11 9.84 1.33 22.46 3.45 26.75 6.03l-2.06 3.43c-2.84-1.71-12.03-3.71-25.22-5.49-4.11-.55-7.08-4.1-6.9-8.25.1-2.17 1.04-4.16 2.65-5.61 1.59-1.43 3.64-2.14 5.77-2.01 7.67.48 18.97 3.05 25.23 4.59-1.02-3.37-.72-7.37 1.07-10.42 1.36-2.31 5.61-7.37 15.74-4.31 15.87 4.78 58.09 55.48 59.88 57.64l-3.08 2.55v.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M140.22 166.97c-2.3-1.73-10.9-2.99-22.99-3.37-5.84-.18-10.47-5.09-10.32-10.94.11-4.48 3.04-8.44 7.28-9.87l2.11-.71 1.27 3.79-2.11.71c-2.66.89-4.49 3.37-4.55 6.17-.09 3.66 2.8 6.73 6.45 6.85 13.65.43 22.16 1.83 25.27 4.17l-2.4 3.2h-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M180.309 226.14c-16.99-25.12-39-41.13-65.42-47.6l-.5-.12c-2.85-.7-5.15-2.77-6.15-5.53-1.04-2.86-.5-6.12 1.4-8.5l3.29-4.1 3.12 2.5-3.29 4.1c-1.04 1.3-1.33 3.07-.76 4.64.55 1.51 1.8 2.63 3.35 3.02l.5.12c27.41 6.71 50.22 23.28 67.78 49.25l-3.31 2.24-.01-.02ZM160.152 143.719l-2.585 3.053 8.021 6.792 2.585-3.052-8.021-6.793Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M168.409 172.4c-6.01-10.42-2.18-27.59-2.01-28.32l3.9.89c-.04.16-3.65 16.37 1.58 25.42l-3.46 2-.01.01ZM147.883 144.507l-12.574 27.227 3.631 1.677 12.575-27.227-3.632-1.677Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M76.52 132.04c32.872 0 59.52-26.648 59.52-59.52S109.392 13 76.52 13 17 39.648 17 72.52s26.648 59.52 59.52 59.52Z"
              fill={
                {
                  "payload": 4289392367,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M73.4798 108.63c2.5626 0 4.64-2.077 4.64-4.64 0-2.563-2.0774-4.64-4.64-4.64-2.5626 0-4.64 2.077-4.64 4.64 0 2.563 2.0774 4.64 4.64 4.64ZM73.24 92.86c-.83 0-1.5-.67-1.5-1.5 0-18.43 5.9-22.29 12.15-26.38.95-.62 1.93-1.26 2.88-1.96 4.85-3.55 6.86-9.58 5.11-15.38-1.34-4.42-5.25-9.26-12.22-9.58h-.26c-.24 0-.54.01-.88-.04-4.42-.71-11.67 1.68-15.44 6.38-2.86 3.57-3.33 7.82-1.41 12.63.31.77-.07 1.64-.84 1.95-.77.31-1.64-.07-1.95-.84-2.93-7.34-.61-12.54 1.85-15.62 4.57-5.7 13.04-8.31 18.26-7.46.11.02.22 0 .33 0h.47c7.14.32 12.87 4.81 14.96 11.71 2.12 7.03-.32 14.36-6.22 18.67-1.01.74-2.03 1.4-3.01 2.05-5.79 3.79-10.79 7.06-10.79 23.87 0 .83-.67 1.5-1.5 1.5h.01Z"
              fill={
                {
                  "payload": 4278240714,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "height": 24,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        dynamicTypeRamp="title2"
        maxFontSizeMultiplier={1.25}
        style={
          [
            {},
            {
              "color": "#0E0F13",
              "fontFamily": "Titillium Sans Pro",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "600",
              "lineHeight": 33,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Can't log in with CieID?
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <Text
        allowFontScaling={false}
        dynamicTypeRamp="body"
        maxFontSizeMultiplier={1.25}
        style={
          [
            {},
            {
              "color": "#555C70",
              "fontFamily": "Titillium Sans Pro",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        If you have SPID, log in to IO by entering your credentials. You can also use your face or fingerprint or the code set for your SPID provider's app.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Login with SPID"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            testID="cie-id-error-primary-action"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 4,
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 16,
                    "textAlignVertical": "center",
                  },
                  {
                    "overflow": "hidden",
                  },
                  false,
                  {
                    "height": 40,
                  },
                  {
                    "backgroundColor": "#0073E6",
                  },
                  {
                    "backgroundColor": undefined,
                    "transform": [
                      {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    false,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessible={false}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  importantForAccessibility="no-hide-descendants"
                  maxFontSizeMultiplier={1.25}
                  numberOfLines={1}
                  style={
                    [
                      {},
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "Titillium Sans Pro",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "lineHeight": 20,
                      },
                      {
                        "alignSelf": "center",
                      },
                    ]
                  }
                >
                  Login with SPID
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Close"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 14,
                "left": 24,
                "right": 24,
                "top": 14,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTouchEnd={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            testID="cie-id-error-secondary-action"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "textAlignVertical": "center",
                  },
                  false,
                  {},
                  {
                    "transform": [
                      {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                accessible={false}
                allowFontScaling={false}
                ellipsizeMode="tail"
                importantForAccessibility="no-hide-descendants"
                maxFontSizeMultiplier={1.25}
                numberOfLines={1}
                style={
                  [
                    {},
                    {
                      "color": "#0E0F13",
                      "fontFamily": "Titillium Sans Pro",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "lineHeight": undefined,
                    },
                    {
                      "color": undefined,
                    },
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`CieIdErrorScreen where device supports NFC Should match the snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flexGrow": 1,
      "marginHorizontal": 24,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
      ]
    }
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      [
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={120}
          bbWidth={120}
          color="#00C5CA"
          focusable={false}
          height={120}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 120,
                "width": 120,
              },
            ]
          }
          tintColor="#00C5CA"
          vbHeight={240}
          vbWidth={240}
          width={120}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="M220.46 181.2c-.11-.13-10.98-13.21-23.59-26.68-20.84-22.26-30.19-28.43-34.36-29.69-5.3-1.59-9.25-.71-11.14 2.51-1.96 3.35-1.13 8.07.92 10.06.66.63.8 1.63.35 2.42-.45.79-1.38 1.18-2.26.94-.18-.05-18.54-4.97-29.01-5.63-1.05-.07-2.06.29-2.85.99-.81.72-1.28 1.72-1.32 2.81-.09 2.07 1.39 3.83 3.43 4.11 9.84 1.33 22.46 3.45 26.75 6.03l-2.06 3.43c-2.84-1.71-12.03-3.71-25.22-5.49-4.11-.55-7.08-4.1-6.9-8.25.1-2.17 1.04-4.16 2.65-5.61 1.59-1.43 3.64-2.14 5.77-2.01 7.67.48 18.97 3.05 25.23 4.59-1.02-3.37-.72-7.37 1.07-10.42 1.36-2.31 5.61-7.37 15.74-4.31 15.87 4.78 58.09 55.48 59.88 57.64l-3.08 2.55v.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M140.22 166.97c-2.3-1.73-10.9-2.99-22.99-3.37-5.84-.18-10.47-5.09-10.32-10.94.11-4.48 3.04-8.44 7.28-9.87l2.11-.71 1.27 3.79-2.11.71c-2.66.89-4.49 3.37-4.55 6.17-.09 3.66 2.8 6.73 6.45 6.85 13.65.43 22.16 1.83 25.27 4.17l-2.4 3.2h-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M180.309 226.14c-16.99-25.12-39-41.13-65.42-47.6l-.5-.12c-2.85-.7-5.15-2.77-6.15-5.53-1.04-2.86-.5-6.12 1.4-8.5l3.29-4.1 3.12 2.5-3.29 4.1c-1.04 1.3-1.33 3.07-.76 4.64.55 1.51 1.8 2.63 3.35 3.02l.5.12c27.41 6.71 50.22 23.28 67.78 49.25l-3.31 2.24-.01-.02ZM160.152 143.719l-2.585 3.053 8.021 6.792 2.585-3.052-8.021-6.793Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M168.409 172.4c-6.01-10.42-2.18-27.59-2.01-28.32l3.9.89c-.04.16-3.65 16.37 1.58 25.42l-3.46 2-.01.01ZM147.883 144.507l-12.574 27.227 3.631 1.677 12.575-27.227-3.632-1.677Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M76.52 132.04c32.872 0 59.52-26.648 59.52-59.52S109.392 13 76.52 13 17 39.648 17 72.52s26.648 59.52 59.52 59.52Z"
              fill={
                {
                  "payload": 4289392367,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M73.4798 108.63c2.5626 0 4.64-2.077 4.64-4.64 0-2.563-2.0774-4.64-4.64-4.64-2.5626 0-4.64 2.077-4.64 4.64 0 2.563 2.0774 4.64 4.64 4.64ZM73.24 92.86c-.83 0-1.5-.67-1.5-1.5 0-18.43 5.9-22.29 12.15-26.38.95-.62 1.93-1.26 2.88-1.96 4.85-3.55 6.86-9.58 5.11-15.38-1.34-4.42-5.25-9.26-12.22-9.58h-.26c-.24 0-.54.01-.88-.04-4.42-.71-11.67 1.68-15.44 6.38-2.86 3.57-3.33 7.82-1.41 12.63.31.77-.07 1.64-.84 1.95-.77.31-1.64-.07-1.95-.84-2.93-7.34-.61-12.54 1.85-15.62 4.57-5.7 13.04-8.31 18.26-7.46.11.02.22 0 .33 0h.47c7.14.32 12.87 4.81 14.96 11.71 2.12 7.03-.32 14.36-6.22 18.67-1.01.74-2.03 1.4-3.01 2.05-5.79 3.79-10.79 7.06-10.79 23.87 0 .83-.67 1.5-1.5 1.5h.01Z"
              fill={
                {
                  "payload": 4278240714,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "height": 24,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        dynamicTypeRamp="title2"
        maxFontSizeMultiplier={1.25}
        style={
          [
            {},
            {
              "color": "#0E0F13",
              "fontFamily": "Titillium Sans Pro",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "600",
              "lineHeight": 33,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Can't log in with CieID?
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <Text
        allowFontScaling={false}
        dynamicTypeRamp="body"
        maxFontSizeMultiplier={1.25}
        style={
          [
            {},
            {
              "color": "#555C70",
              "fontFamily": "Titillium Sans Pro",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        If you have your Electronic Identity Card handy, use it to access IO.

Insert the CIE PIN and bring it close to the device to allow the data to be read.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Login with CIE + PIN"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            testID="cie-id-error-primary-action"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 4,
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 16,
                    "textAlignVertical": "center",
                  },
                  {
                    "overflow": "hidden",
                  },
                  false,
                  {
                    "height": 40,
                  },
                  {
                    "backgroundColor": "#0073E6",
                  },
                  {
                    "backgroundColor": undefined,
                    "transform": [
                      {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    false,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessible={false}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  importantForAccessibility="no-hide-descendants"
                  maxFontSizeMultiplier={1.25}
                  numberOfLines={1}
                  style={
                    [
                      {},
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "Titillium Sans Pro",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "lineHeight": 20,
                      },
                      {
                        "alignSelf": "center",
                      },
                    ]
                  }
                >
                  Login with CIE + PIN
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Close"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 14,
                "left": 24,
                "right": 24,
                "top": 14,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTouchEnd={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            testID="cie-id-error-secondary-action"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "textAlignVertical": "center",
                  },
                  false,
                  {},
                  {
                    "transform": [
                      {
                        "scale": undefined,
                      },
                    ],
                  },
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                accessible={false}
                allowFontScaling={false}
                ellipsizeMode="tail"
                importantForAccessibility="no-hide-descendants"
                maxFontSizeMultiplier={1.25}
                numberOfLines={1}
                style={
                  [
                    {},
                    {
                      "color": "#0E0F13",
                      "fontFamily": "Titillium Sans Pro",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "lineHeight": undefined,
                    },
                    {
                      "color": undefined,
                    },
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
